### PR TITLE
Add relationship evaluation response schema

### PR DESCRIPTION
**Pull request by codeSafari10**
**Notes for Reviewers**

This PR fixes #

Adds schema for the tier 2 evaluation. 

# Relation Policy Evaluation

## Input
- Design file to evaluate
- Contains: components and relationships
- Relationships can have status for tracking between different evaluation phases:
  - pending
  - valid
  - invalid

### Phase One: Feasibility Check (tier one) 
- Input: [Component], [Design]
- Output: [Relationships] (can be scoped to particular kind, type, subtype)
- Purpose: Determine possible relationships that can be created
- Result: List of pending relationships that can be created

### Phase One : Configure 
- Input: All pending relationships
- Action: Complete relationships by patching the fields

### Phase two : Match
- Input: [Design]
- Output: [Relationships]
- Purpose: 
  1. Find all valid relationships present in the design
  2. Validate existing relationships
  3. Infer new relationships that can be created

### Phase three : Action
- Input: All valid relationships
- Action: Perform actions on relationships, such as:
  - Patching the configuration
  - Creating subsidiary components
  - Removing components

## Response
The evaluator takes a design file with possible pending relationships and returns:
1. The diff, or
2. Complete updated design file with all relationships patched and updated

### Additional Response Options
Return a traceback of operations performed, for example:

```json
{
  "policyActions": [],
  "relationshipsAdded": [],
  "relationshipsRemoved": [],
  "componentsAdded": [],
  "componentsRemoved": []
} 

```

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: schemas/constructs/openapi/relationshipEvaluation.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+openapi: 3.0.0
```
Comment (leecalcote) [inline schemas/constructs/openapi/relationshipEvaluation.yml:1]: Should we be using this be 3.1.0?

File: schemas/constructs/openapi/relationshipEvaluation.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+info:
```
Comment (leecalcote) [inline schemas/constructs/openapi/relationshipEvaluation.yml:2]: Should this construct be defined generically, looking ahead to capabilitiesEvalutation? @codeSafari10 @MUzairS15 @aabidsofi19 @humblenginr

File: schemas/constructs/v1alpha1/relationshipEvaluation.json
Comment (leecalcote) [inline schemas/constructs/v1alpha1/relationshipEvaluation.json:1]: Should this construct be defined generically, looking ahead to capabilitiesEvalutation? @codeSafari10 @MUzairS15 @aabidsofi19 @humblenginr

File: schemas/constructs/v1alpha1/relationshipEvaluation.json [hunk 3]
```diff
@@ -0,0 +1,88 @@
+{
+  "$id": "https://schemas.meshery.io/evaluationResponse.json",
```
Comment (leecalcote) [inline schemas/constructs/v1alpha1/relationshipEvaluation.json:2]: The intermixing of OpenAPI 3.1.0 above and JSON Schema draft 7 here confuses me a bit.

File: schemas/constructs/v1alpha1/relationshipEvaluation.json [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    "evaluationHash": {
+      "type": "string",
+      "description": "Hash of the evaluation response, used for tracking and comparison."
+    },
+
+    "metadata": {
+      "type": "object",
+      "description": "Additional metadata about the evaluation process.",
+      "properties": {
+        "timestamp": {
```
Comment (leecalcote) [inline schemas/constructs/v1alpha1/relationshipEvaluation.json:78]: I support either promoting timestamp to a top-level property or pushing `evaluationHash` into `metadata`.

File: schemas/constructs/openapi/relationshipEvaluation.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+          description: Successful evaluation
+          content:
+            application/json:
+              schema:
+                $ref: "https://schemas.meshery.io/evaluationResponse.json"
+        "400":
+          description: Bad request
+        "500":
+          description: Internal server error
+
```
Comment (leecalcote) [inline schemas/constructs/openapi/relationshipEvaluation.yml:29]: Would dry-run ever be useful?

While GETting of prior eval results is not to supported by this API, each evaluation result should be sent as an event to Meshery Server, and if the provider supports it, the provider can persist that event.

File: schemas/constructs/openapi/relationshipEvaluation.yml (hunk 1 above)
Comment (leecalcote) [reply schemas/constructs/openapi/relationshipEvaluation.yml:1]: @MUzairS15 to confirm whether oapi-codegen supports the 3.1.0 draft. By default, it doesn't.

File: schemas/constructs/v1alpha1/relationshipEvaluation.json (hunk 3 above)
Comment (aabidsofi19) [reply schemas/constructs/v1alpha1/relationshipEvaluation.json:2]: yeah , makes sense to me to have it as openapi schema.

Comment (aabidsofi19) [reply schemas/constructs/v1alpha1/relationshipEvaluation.json:2]: we need stricter guidelines to where which types of schema's belong

File: schemas/constructs/openapi/relationshipEvaluation.yml (hunk 1 above)
Comment (aabidsofi19) [reply schemas/constructs/openapi/relationshipEvaluation.yml:1]: other schemas are also using openapi 3.0.0 right now

File: schemas/constructs/v1alpha1/relationshipEvaluation.json (hunk 3 above)
Comment (MUzairS15) [reply schemas/constructs/v1alpha1/relationshipEvaluation.json]: I propose to migrate all schemas to be in openapi format.

**Pull request merged**